### ISSUE TITLE
Ask for some help

### ISSUE THREAD

**Issue by hhhmoan** [qingzew/tensorflow-ram#1]
thanks for release your code. now i just want to re-implement the RAM in tensorflow. but it always get bad result. can you code  achieve the accuary release by the paper “Recurrent models of Visual attention”?

**Comment by wqingzex** [comment]
No, I didn't, maybe because I didn't use rnn.

**Comment by hhhmoan** [comment]
thanks for your answer,but why you use your grad_reinforce function but not just 
tf.graident((l_mean-l_sample)**2/balabala  \* (R-tf.reduce_mean(R))). Is there difference?
and in your ram.py the line 294, there really need a nagetive?

**Comment by wqingzex** [comment]
@hhhmoan  the method is called "Policy gradient methods", one implementation of attention model, you can read [this](http://www.scholarpedia.org/article/Policy_gradient_methods#Likelihood_Ratio_Methods_and_REINFORCE) for more details, another implementation can be found in [here](https://blog.heuritech.com/2016/01/20/attention-mechanism/)

maybe, I was wrong, if so ,please tell me

**Comment by hhhmoan** [comment]
when you train your model in translate clutter mnist, how much image do you use? and your gradient method is right,but in tensorflow, maybe you code can be more easier. and in tensorflow , the randomnormal function also have gradients,maybe you should use stop_gradient. and really thank you release your code .

**Comment by wqingzex** [comment]
@hhhmoan  my colleague share the training data with me, ~5k per digit.

do you mean `tf.random_normal` function? although the function also gradients, but i think only the trainable variables need to be compute, so if you have some good ideas, please share it with me, also including the `stop_gradient` function

**Comment by hhhmoan** [comment]
yes, Maybe you are right .And,could your release the code in translate clutter mnist，i want to try it by myself.If I've been a little abrupt, then I sincerely apologize. thank you

**Comment by wqingzex** [comment]
I test these codes on dataset of translate clutter mnist, but with size 40 \* 40, and also original mnist dataset, it's ok. If something is wrong, please tell me.

By the way, I convert the original dataset file to images per digit

**Comment by hhhmoan** [comment]
maybe you should test the code in translate clutter mnist 100_100 with depth 3 and glimpse 12_12,because my code also can make ok in small image, but when in 100*100,it's fail,if you need data,i can share with you

**Comment by wqingzex** [comment]
really? if true, it's my fault. please share the dataset with me, I will try it.

**Comment by wqingzex** [comment]
did you try different parameters? maybe it's not the right configuration. what's your problem

**Comment by hhhmoan** [comment]
let me check my data，and share you latter

**Comment by hhhmoan** [comment]
the num of image is really too much, so i'd like to give you the script to get them.  https://github.com/deepmind/mnist-cluttered   if you finfish your code, please tell me . and you can see my easy code in https://github.com/hhhmoan/RAM-tensorflow

**Comment by hhhmoan** [comment]
have you finish this experiment

**Comment by wqingzex** [comment]
I am not  familiar with lua, so I do not know how to generate a image with 3 channels, did you try decrease the glimpse? maybe 8 would be ok?

**Comment by hhhmoan** [comment]
maybe you can get the grey image and then use python cv2 script to transfer to 3 channel,and i think decrease the number of glimpse is no use, follow the paper's experiment may be better. Thank you for taking time to discuss with me
